### PR TITLE
feat: add release workflow for burrito

### LINKED ISSUE #13: Github Action for executable binary compilation
- With [burrito](https://github.com/burrito-elixir/burrito)
- (optional) apply config from command line args and/or config file

### PR DESCRIPTION
**Pull request by J0**
## What kind of change does this PR introduce?

Drive by contribution since the workflow seems similar to what we have in Auth. Aims to fix #13 , let me know if anything should be amended

How this was tested:

- [x] Tested via [Act](https://github.com/nektos/act) on a [fork](https://github.com/J0/supavisor/actions/runs/5568478329)

## Additional Notes

The action needs to be granted write permissions to  upload the binaries - Settings > Actions  > Workflow Permissions > Read and Write Permissions

### CODE REVIEW

**Comment by chasers** [comment]
Super helpful thanks @J0! Curious, was the binary building for you? 

@abc3 and I do not love the tagging you have to do in commits with this semver action.

I was going to try automating release notes with Github and pulling the version from the mix.exs file. Or pulling the version from a VERSION file into mix.exs and have the Github stuff run off the VERSION file. 

Logflare has [a VERSION file](https://github.com/Logflare/logflare/blob/staging/VERSION) we do things with.

Realtime has a [mix version check](https://github.com/supabase/realtime/blob/main/.github/workflows/version_updated.yml) so it reminds us to update the version there too.

Maybe you can break this PR up into two? One that gets the binary building on releases. And then we can setup a draft one to flesh out the semver release workflow when @abc3 is back next week? 

FYI, we are also wanting to setup multi-stage deployments with Actions like: https://www.youtube.com/watch?v=sVl6De94evo

**Comment by J0** [comment]
Hey @chasers,

Thanks for the swift reply! Sorry for the delayed response - it's been a busy week. I'll look at all the Github Actions related issues at one go tomorrow.

**Comment by J0** [comment]
Seems like the binary does not build for me. Running into an [error on the build](https://github.com/J0/supavisor/actions/runs/4844468261/jobs/8632763390#step:7:210) when [replacing ERTS binaries]((https://github.com/burrito-elixir/burrito/blob/main/lib/steps/patch/copy_erts.ex#L46) - if anyone has suggestions do let me know.

Reverting to draft for now.

**Comment by J0** [comment]
Opened an issue here: https://github.com/burrito-elixir/burrito/issues/90

**Comment by J0** [comment]
Going to close for now until the linked issue is resolved

**Comment by J0** [comment]
Hey team,

This took quite.a while - had to resolve some dependency issues. Was hoping to close this out - wondering if y'all could take a look  whenever a slot frees up and see if this is what y'all had in mind

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+          elixir-version: 1.14.x # Define the elixir version [required]
+          experimental-otp: true
+      - name: Cache Mix
+        uses: actions/cache@v3
+        with:
+          path: deps
+          key: ${{ runner.os }}-mix-${{ hashFiles(format('{0}{1}', github.workspace, '/mix.lock')) }}
+          restore-keys: |
+            ${{ runner.os }}-mix-
+      - name: Cache Burrito OTP  deps
```
Comment (J0) [inline .github/workflows/release.yml:27]: So that we don't pose too much of a load on Burrito's CDN when pulling Erlang and OTP deps

**Comment by J0** [comment]
Hey @chasers @abc3,

Not urgent, but just bumping in case it was missed - could I trouble y'all for a review on this whenever free?  Let me know!

**Comment by J0** [comment]
Hey @chasers @abc3,

I'm going to close this for now as it's been sitting for a while but let me know if it's ever relevant in the future and we can re-open.

Thanks!

**Review by chasers — APPROVED** [review]

**Review by chasers — APPROVED** [review]

**Comment by J0** [comment]
As discussed internally, let's table this for now

**Pull request closed without merging**